hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/realtime/RealtimeUnmergedRecordReader.java

```diff
@@ -82,7 +82,7 @@ public RealtimeUnmergedRecordReader(HoodieRealtimeFileSplit split, JobConf job,
         false, jobConf.getInt(MAX_DFS_STREAM_BUFFER_SIZE_PROP, DEFAULT_MAX_DFS_STREAM_BUFFER_SIZE), record -> {
           // convert Hoodie log record to Hadoop AvroWritable and buffer
           GenericRecord rec = (GenericRecord) record.getData().getInsertValue(getReaderSchema()).get();
-          ArrayWritable aWritable = (ArrayWritable) avroToArrayWritable(rec, getWriterSchema());
+          ArrayWritable aWritable = (ArrayWritable) avroToArrayWritable(rec, getHiveSchema());
```

Comment:
what do we use in the merged record reader? cc @bvaradar

Reply:
@vinothchandar This is a bug when using RealtimeUnmergedRecordReader, merged record reader works fine. unmerged record reader should use hive schema rather than writer schema here.

Reply:
sg.